src/hl/datatype.rs

```diff
@@ -261,7 +261,7 @@ impl Datatype {
                     let mut fields: Vec<CompoundField> = Vec::new();
                     for idx in 0..h5try!(H5Tget_nmembers(id)) as _ {
                         let name = H5Tget_member_name(id, idx);
-                        let offset = h5try!(H5Tget_member_offset(id, idx));
+                        let offset = H5Tget_member_offset(id, idx);
```

Comment:
Hmm...

> Returns the byte offset of the field if successful; otherwise returns 0 (zero). Note that zero is a valid offset and that this function will fail only if a call to H5Tget_member_class() fails with the same arguments.

Indeed, looks like it can't fail. If there's something wrong, then it's the line below (assigning `ty`) that will fail.

Reply:
Exactly the same for `H5Tget_size`, but I could not make a test with a zero-sized type. Tried with a `[0; u8]`, but empty fixed arrays are not legal.